# HG changeset patch
# User Neil <nyamatongwe@gmail.com>
# Date 1492232428 -36000
# Node ID b8379bec8e70bf09df25f01d0a78a53cf61b1827
# Parent  692a54eaa6049d6ea3459f66c001ef0c8ea7668f
Fix deprecation of gdk_window_process_updates in GTK+ 3.22 by redrawing instead
of scrolling.

diff -r 692a54eaa604 -r b8379bec8e70 gtk/ScintillaGTK.cxx
--- a/gtk/ScintillaGTK.cxx	Sat Apr 15 14:28:09 2017 +1000
+++ b/gtk/ScintillaGTK.cxx	Sat Apr 15 15:00:28 2017 +1000
@@ -1031,16 +1031,18 @@
 }
 
 void ScintillaGTK::ScrollText(Sci::Line linesToMove) {
-	int diff = vs.lineHeight * -linesToMove;
-	//Platform::DebugPrintf("ScintillaGTK::ScrollText %d %d %0d,%0d %0d,%0d\n", linesToMove, diff,
-	//	rc.left, rc.top, rc.right, rc.bottom);
+	NotifyUpdateUI();
+
+#if GTK_CHECK_VERSION(3,22,0)
+	Redraw();
+#else
 	GtkWidget *wi = PWidget(wText);
-	NotifyUpdateUI();
-
 	if (IS_WIDGET_REALIZED(wi)) {
+		const int diff = vs.lineHeight * -linesToMove;
 		gdk_window_scroll(WindowFromWidget(wi), 0, -diff);
 		gdk_window_process_updates(WindowFromWidget(wi), FALSE);
 	}
+#endif
 }
 
 void ScintillaGTK::SetVerticalScrollPos() {
